### PR TITLE
feat: Add phone number step to Submission artwork form

### PR DESCRIPTION
**Pull request by olerichter00**
This PR resolves [ONYX-1011] 

### Description

This PR adds the "Add phone number" step to the new Submission artwork form. The phone number will not be validated and is not required.

| | |
| --- | --- |
|Simulator Screenshot - iPhone 15 Pro - 2024-05-30 at 16 17 05 |Simulator Screenshot - iPhone 15 Pro - 2024-05-30 at 16 17 12 |

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [ ] I hid my changes behind a **[feature flag]**, or they don't need one.
- [ ] I have included **screenshots** or **videos**, or I have not changed the UI.
- [ ] I have added **tests**, or my changes don't require any.
- [ ] I added an **[app state migration]**, or my changes do not require one.
- [ ] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [ ] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Add phone number step to new SWA artwork form (behind ff)- ole

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-1011]: https://artsyproduct.atlassian.net/browse/ONYX-1011?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Components/Input/PhoneInput/PhoneInput.tsx [hunk 1]
```diff
@@ -23,7 +23,7 @@ import { formatPhoneNumber } from "./formatPhoneNumber"
 export const PhoneInput = forwardRef<
   InputRef,
   {
-    setValidation: (value: boolean) => void
+    setValidation?: (value: boolean) => void
```
Comment (olerichter00) [inline src/app/Components/Input/PhoneInput/PhoneInput.tsx:26]: This prop should be optional because it's not always needed.

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 61415b5ea574bdeed4e0d8cb8c0a231b99eea094
</p>

File: src/app/Scenes/SellWithArtsy/ArtworkForm/Utils/navigationHelpers.ts [hunk 2]
```diff
@@ -39,15 +41,20 @@ export const useSubmissionContext = () => {
 
       const newValues = {
         ...values,
-        state: (currentStep === "AddDimensions"
-          ? "SUBMITTED"
-          : undefined) as ArtworkDetailsFormModel["state"],
+        state: (isLastStep ? "SUBMITTED" : undefined) as ArtworkDetailsFormModel["state"],
```
Comment (olerichter00) [inline src/app/Scenes/SellWithArtsy/ArtworkForm/Utils/navigationHelpers.ts:44]: Setting the state to "Submitted" in the "AddDimensions" step breaks the form if it is not the last step. 

With this change, the last step is defined correctly as the final step in `ARTWORK_FORM_STEPS`.

File: src/app/Scenes/SellWithArtsy/ArtworkForm/Utils/navigationHelpers.ts [hunk 3]
```diff
@@ -39,15 +41,20 @@ export const useSubmissionContext = () => {
...
+        try {
+          const submissionId = await createOrUpdateSubmission(newValues, values.submissionId)
+
+          if (!values.submissionId && submissionId) {
+            setFieldValue("submissionId", submissionId)
+          }
+        } catch (error) {
+          console.error("Error creating or updating submission", error)
+          Alert.alert("Could not create or update submission")
+          return
```
Comment (olerichter00) [inline src/app/Scenes/SellWithArtsy/ArtworkForm/Utils/navigationHelpers.ts]: We should show an error message when the submission fails to update.

File: src/app/Scenes/SellWithArtsy/ArtworkForm/Utils/validation.ts [hunk 4]
```diff
@@ -31,12 +31,7 @@ export const getCurrentValidationSchema = (_injectedStep?: keyof SubmitArtworkSt
       return provenanceSchema
     case "AddDimensions":
       return dimensionsSchema
-
     default:
-      // Make sure the devs are warned when they forget to add a validation schema
-      if (currentStep && currentStep !== "StartFlow") {
-        console.warn(`No validation schema found for step: ${currentStep}`)
-      }
```
Comment (olerichter00) [inline src/app/Scenes/SellWithArtsy/ArtworkForm/Utils/validation.ts:39]: In my opinion, this isn't needed, and we can simply default to no validation for steps that don't require validation.

**Review by dariakoko — APPROVED** [review]

**Pull request merged**